pkg/redis/RedisQueueConsumer.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Enqueue\Redis;
+
+class RedisQueueConsumer
+{
+    /**
+     * @var Redis
+     */
+    private $redis;
+
+    /**
+     * @var RedisDestination[]
+     */
+    private $queues;
+
+    /**
+     * Return back message into the queue if message was not acknowledged or rejected after this delay.
+     * It could happen if consumer has failed with fatal error or even if message processing is slow
+     * and takes more than this time.
+     *
+     * @var int
+     */
+    private $retryDelay = 300;
+
+    /**
+     * @var string[]
+     */
+    private $queueNames;
+
+    /**
+     * @param Redis              $redis
+     * @param RedisDestination[] $queues
+     */
+    public function __construct(Redis $redis, array $queues)
```

Comment:
I'd move `array $queues` to receiveXXX methods. I think it would be better to make it as stateless as possible